### PR TITLE
adds random button

### LINKED ISSUE #911: reinstate the random question button
In the old site there was a dice button in the top right of a question page which sent you to a random open question.

### PR DESCRIPTION
**Pull request by lsabor**
closes #911 

Adds random question button to the navbar. :)

image

### CODE REVIEW

File: posts/views.py [hunk 1]
```diff
@@ -678,3 +678,15 @@ def download_csv(request, pk: int):
...
+@api_view(["GET"])
+@permission_classes([AllowAny])
+def random_post_id(request):
+    post = (
+        Post.objects.filter_permission(user=request.user)
+        .filter_active()
+        .order_by("?")
+        .first()
+    )
+    return Response({"id": post.id, "url_title": post.url_title})
```
Comment (hlbmtc) [inline posts/views.py:692]: `url_title` is not very safe to use since it contains spaces, special chars and etc, please use `get_post_slug` instead

Comment (hlbmtc) [reply posts/views.py:692]: Also, we have different urls for questions from communities and notebooks (see `getPostLink` function on the frontend)

File: front_end/src/components/random_button.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+const RandomButton: FC = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const router = useRouter();
+
+  const handleRandomClick = async () => {
+    setIsLoading(true);
+    try {
+      const data = await fetchRandomPostId();
+      router.push(`/questions/${data.id}/${data.url_title}/`);
```
Comment (hlbmtc) [inline front_end/src/components/random_button.tsx:20]: We have different urls for questions from communities and notebooks, so check `getPostLink` function (probably we'll need to update endpoint to return all required post data for it).

Alternatively, you can narrow down Post response to questions only and not to overload BE response with extra params

File: front_end/src/components/icons/die.css [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        transform: rotate(0deg);
+    }
+    90% {
+        transform: rotate(360deg);
+    }
+    100% {
+        transform: rotate(360deg);
+    }
+}
+
```
Comment (hlbmtc) [inline front_end/src/components/icons/die.css:29]: EOF

File: front_end/src/components/icons/die.css
Comment (hlbmtc) [inline front_end/src/components/icons/die.css:1]: Can we replace this with pure tailwind?

File: front_end/src/components/random_button.tsx (hunk 2 above)
Comment (lsabor) [reply front_end/src/components/random_button.tsx:20]: Oh, right. I'll restrict the endpoint to only Question type Posts.

File: front_end/src/components/icons/die.css
Comment (lsabor) [reply front_end/src/components/icons/die.css:1]: @ncarazon I'd need your help on that.
Or I can move the css clsees to within the random_button.tsx file.

Comment (lsabor) [reply front_end/src/components/icons/die.css:1]: I think I figured it out, but my change definitely needs to be looked over.

File: front_end/src/components/random_button.tsx [hunk 4]
```diff
@@ -31,7 +30,8 @@ const RandomButton: FC = () => {
       disabled={isLoading}
       aria-label="Random Question"
       style={{ background: "transparent" }}
```
Comment (hlbmtc) [inline front_end/src/components/random_button.tsx:32]: @lsabor Remove this style and just replace with native `<button>` -- we don't need our specific UI button here

File: front_end/src/components/icons/die.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+export function Die({ className, ...props }: React.SVGProps<SVGSVGElement>) {
+  const diePath =
+    "M64 80c-8.8 0-16 7.2-16 16V416c0 8.8 7.2 16 16 16H384c8.8 " +
+    "0 16-7.2 16-16V96c0-8.8-7.2-16-16-16H64zM0 96C0 60.7 28.7 32 " +
+    "64 32H384c35.3 0 64 28.7 64 64V416c0 35.3-28.7 64-64 64H64c-35.3 " +
+    "0-64-28.7-64-64V96zm128 32a32 32 0 1 1 0 64 32 32 0 1 1 0-64zm0 " +
+    "192a32 32 0 1 1 0 64 32 32 0 1 1 0-64zm64-64a32 32 0 1 1 64 0 32 32 " +
+    "0 1 1 -64 0zM320 128a32 32 0 1 1 0 64 32 32 0 1 1 0-64zM288 352a32 32 " +
+    "0 1 1 64 0 32 32 0 1 1 -64 0z";
```
Comment (ncarazon) [inline front_end/src/components/icons/die.tsx:11]: nit: define this outside of the component, so `diePath` variable is not re-initialized on every re-render:

```
const `DIE_PATH = ""

export function Die ...
```

**Review by ncarazon — APPROVED** [review]

**Pull request merged**